frontend/src/components/StatsCard.js

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from "react";
+import { Stack, Text, chakra, Box, SimpleGrid, Link } from "@chakra-ui/react";
+import { TriangleDownIcon, TriangleUpIcon } from "@chakra-ui/icons";
+import useNFTs from "../core/hooks/useNFTs";
+import web3 from "web3";
+
+const TIME_PERIOD = {
+  current: 0,
+  previous: 1,
+};
+
+const isNumberNotZero = (str) => {
+  if (isNaN(Number(str) || Number(str) == 0)) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+const getEthValue = (string) => {
+  const ether = web3.utils.fromWei(string, "ether");
+  return nFormatter(ether, 3);
+};
+
+const nFormatter = (num, digits) => {
+  const lookup = [
+    { value: 1, symbol: "" },
+    { value: 1e3, symbol: "k" },
+    { value: 1e6, symbol: "M" },
+    { value: 1e9, symbol: "G" },
+    { value: 1e12, symbol: "T" },
+    { value: 1e15, symbol: "P" },
+    { value: 1e18, symbol: "E" },
+  ];
+  const rx = /\.0+$|(\.[0-9]*[1-9])0+$/;
+  var item = lookup
+    .slice()
+    .reverse()
+    .find(function (item) {
+      return num >= item.value;
+    });
+  return item
+    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
```

Comment:
Can you do this using numerical operations rather than string operations?

Reply:
I really don't like use of regex for this kind of thing. If you must use it, put a comment explaining what the regex is doing with some examples.

Reply:
Most likely, you do not have to use regex.

Reply:
This regex part is to remove trailing zero. 

I decided to remove it at all as it keeps things clear on decimal parts. And also use 2 fractional digits instead of 3, seems visually more clear then